src/cluster_legacy.c

```diff
@@ -6543,7 +6556,30 @@ unsigned int delKeysInSlot(unsigned int hashslot, int lazy, bool propagate_del,
      * state so that we don't assert in propagateNow(). */
     server.server_del_keys_in_slot = 1;
     unsigned int j = 0;
+    robj *argv[4];
     int before_execution_nesting = server.execution_nesting;
+    enterExecutionUnit(1, 0);
+
+    /* Propagate as a single CLUSTER FLUSHSLOT <slot> ASYNC/SYNC. */
+    if (propagate_del) {
+        enterExecutionUnit(1, 0);
+        argv[0] = shared.cluster;
+        argv[1] = shared.flushslot;
+        argv[2] = createStringObjectFromLongLong(hashslot);
+        argv[3] = lazy ? shared.async : shared.sync;
+        incrRefCount(argv[0]);
+        incrRefCount(argv[1]);
+        incrRefCount(argv[3]);
```

Comment:
alsoPropagate shouldn't decrement the refcount, so these shouldn't be needed.

Reply:
Hmm, yeah I lifted this from `propagateDeletion` which increments the refcount. Maybe this is just an artifact of previous behavior, since I see that it shouldn't be required.

Reply:
https://github.com/valkey-io/valkey/pull/2175 Yeah, it looks like just dead code to me.